Tests/HaCWebsiteLibTests/HaCWebsiteLibTests.swift

```diff
@@ -2,16 +2,94 @@ import XCTest
 @testable import HaCWebsiteLib
 
 class HaCWebsiteLibTests: XCTestCase {
-    func testExample() {
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct results.
-        XCTAssertEqual(societyName(), "Hackers at Cambridge")
-    }
+  enum TestError: Error {
+    case fileNotFound(String), couldNotLoadFile(String)
+  }
+
+  /**
+   * Gets the absolute path to a resource in HaCWebsiteLibTests/Resources directory.
+   * Generates an XCTest failure and throws an error if the resource cannot be found
+   *
+   * - returns:
+   * A String of the absolute path to the resource
+   *
+   * - throws:
+   * An error of type TestError.fileNotFound if the fil cannot be found
+   *
+   * - parameters:
+   *   - at: The path of the desired resource relative to HaCWebsiteLibTests/Resources
+   *   - file: Optional file name for better error reporting
+   *   - line: Optional line number for better error reporting
+   */
+  private func getTestResourcePath(at relativePath: String, file: String = #file, line: UInt = #line) throws -> String {
+    let thisFile = #file
+    let components = thisFile.characters.split(separator: "/").map(String.init)
+    let toTestsDir = components[0 ..< components.count - 1]
+    let filePath = "/" + toTestsDir.joined(separator: "/") + "/Resources/" + relativePath
 
+    let fileManager = FileManager.default
+
+    if fileManager.fileExists(atPath: filePath) {
+      return filePath
+    } else {
+      self.recordFailure(withDescription: "Could not find file at \(filePath)", inFile: file, atLine: line, expected: false)
+      throw TestError.fileNotFound(filePath)
+    }
+  }
 
-    static var allTests : [(String, (HaCWebsiteLibTests) -> () throws -> Void)] {
-        return [
-            ("testExample", testExample),
-        ]
+  /**
+   * Gets the String contents of a UTF8 encoded resource in HaCWebsiteLibTests/Resources directory.
+   * Generates an XCTest failure and throws an error if the resource cannot be loaded
+   *
+   * - returns:
+   * A String of the resource contents
+   *
+   * - throws:
+   * An error of type TestError.couldNotLoadFile if the file cannot be loaded
+   *
+   * - parameters:
+   *   - from: The path of the desired resource relative to HaCWebsiteLibTests/Resources
+   *   - file: Optional file name for better error reporting
+   *   - line: Optional line number for better error reporting
+   */
+  private func getTestResourceString(from relativePath: String, file: String = #file, line: UInt = #line) throws -> String {
+    let path = try getTestResourcePath(at: relativePath, file: file, line: line)
+    do {
+      return try String(contentsOfFile: path, encoding: .utf8)
+    } catch {
+      self.recordFailure(withDescription: "Could not load file at \(path)", inFile: file, atLine: line, expected: false)
+      throw TestError.couldNotLoadFile("path")
     }
+  }
+
+  func testWorkshopParse() throws {
```

Comment:
It does look like a lot but really it's only testing if the mapping to our internal structure works. The rest is handled by libraries. Certainly once we have a more concrete Yaml format it will be good write some more fine-grained tests for this.